src/dotnet/GeneratorAgent/src/Azure.Tools.GeneratorAgent/Security/InputValidator.cs

```diff
@@ -0,0 +1,274 @@
+                    string normalizedPath = validatedPath.Replace('\\', '/');
+                    
+                    return ValidationResult.Valid(normalizedPath);
+                }
+            }
+            catch (Exception ex)
+            {
+                Logger?.LogWarning(ex, "Invalid path format: {Path}", pathTraversalValidation.Value);
+                return ValidationResult.Invalid($"Invalid path format: {ex.Message}");
+            }
+        }
+
+        /// <summary>
+        /// Validates a Git commit ID format.
+        /// </summary>
+        public static ValidationResult ValidateCommitId(string? commitId)
+        {
+            if (string.IsNullOrWhiteSpace(commitId))
+            {
+                return ValidationResult.Valid(string.Empty); // Commit ID is optional
+            }
+
+            if (!CommitIdRegex.IsMatch(commitId))
+            {
+                Logger?.LogWarning("Invalid commit ID format: {CommitId}", commitId);
+                return ValidationResult.Invalid("Commit ID must be 6-40 hexadecimal characters");
+            }
+
+            return ValidationResult.Valid(commitId);
+        }
+
+        /// <summary>
+        /// Validates a directory path for output operations.
+        /// </summary>
+        public static ValidationResult ValidateOutputDirectory(string? path)
+        {
+            // Use centralized path traversal validation
+            ValidationResult pathTraversalValidation = ValidateDirTraversal(path, "Output directory path");
+            if (!pathTraversalValidation.IsValid)
+            {
+                return pathTraversalValidation;
+            }
+
+            try
+            {
+                string fullPath = Path.GetFullPath(pathTraversalValidation.Value);
+                
+                string? parentDir = Path.GetDirectoryName(fullPath);
+                if (!string.IsNullOrEmpty(parentDir) && !Directory.Exists(parentDir))
+                {
+                    return ValidationResult.Invalid($"Parent directory does not exist: {parentDir}");
+                }
+
+                return ValidationResult.Valid(fullPath);
+            }
+            catch (Exception ex)
+            {
+                Logger?.LogWarning(ex, "Invalid output directory path: {Path}", pathTraversalValidation.Value);
+                return ValidationResult.Invalid($"Invalid path format: {ex.Message}");
+            }
+        }
+
+        /// <summary>
+        /// Validates a PowerShell script path for security.
+        /// </summary>
+        public static ValidationResult ValidatePowerShellScriptPath(string scriptPath, string azureSdkPath)
+        {
+            if (string.IsNullOrWhiteSpace(scriptPath))
+            {
+                return ValidationResult.Invalid("PowerShell script path cannot be null or empty");
+            }
+
+            // Ensure it's a PowerShell script
+            if (!scriptPath.EndsWith(".ps1", StringComparison.OrdinalIgnoreCase))
```

Comment:
```suggestion
            if (!string.Equals(Path.GetExtension(scriptPath), ".ps1", StringComparison.OrdinalIgnoreCase))
```

You'll want to use the `Path` built-ins where you can.